diff --git a/src/gpu/ganesh/mtl/GrMtlCaps.mm b/src/gpu/ganesh/mtl/GrMtlCaps.mm
index 8d0bc6e364..1403609f89 100644
--- a/src/gpu/ganesh/mtl/GrMtlCaps.mm
+++ b/src/gpu/ganesh/mtl/GrMtlCaps.mm
@@ -102,7 +102,11 @@
             return true;
         }
     }
-#elif defined(SK_BUILD_FOR_IOS)
+#endif
+
+#if !(defined(TARGET_OS_TV) && TARGET_OS_TV)
+
+#if defined(SK_BUILD_FOR_IOS)
     // TODO: support tvOS
    *gpuFamily = GPUFamily::kApple;
     // iOS 12
@@ -164,6 +168,34 @@
     }
     // We don't support earlier OSes
 #endif
+#else
+    *gpuFamily = GPUFamily::kApple;
+
+    if ([device supportsFeatureSet:MTLFeatureSet_tvOS_GPUFamily2_v2]) {   
+        *group = 2;
+        return true;
+    }
+    if ([device supportsFeatureSet:MTLFeatureSet_tvOS_GPUFamily2_v1]) { 
+        *group = 2;
+        return true;
+    }
+    if ([device supportsFeatureSet:MTLFeatureSet_tvOS_GPUFamily1_v4]) {
+        *group = 1;
+        return true;
+    }
+    if ([device supportsFeatureSet:MTLFeatureSet_tvOS_GPUFamily1_v3]) {
+        *group = 1;   
+        return true;
+    }
+    if ([device supportsFeatureSet:MTLFeatureSet_tvOS_GPUFamily1_v2]) {
+        *group = 1;
+        return true;
+    }
+    if ([device supportsFeatureSet:MTLFeatureSet_tvOS_GPUFamily1_v1]) {
+         *group = 1;
+        return true;
+    }
+#endif //tvos
 
 #endif // GR_METAL_SDK_VERSION < 300
 
@@ -171,9 +203,16 @@
     return false;
 }
 
+#if defined(TARGET_OS_TV) && (__TV_OS_VERSION_MAX_ALLOWED < 130000)
+//#error "__TV_OS_VERSION_MAX_ALLOWED baddd"
+#endif
+
 bool GrMtlCaps::getGPUFamily(id<MTLDevice> device, GPUFamily* gpuFamily, int* group) {
 #if GR_METAL_SDK_VERSION >= 220
-    if (@available(macOS 10.15, iOS 13.0, tvOS 13.0, *)) {
+#if !defined(TARGET_OS_TV) 
+    if (@available(macOS 10.15, iOS 13.0, tvOS 13.0, *))
+#endif	
+	{
         // Apple Silicon
 #if GR_METAL_SDK_VERSION >= 230
         if ([device supportsFamily:MTLGPUFamilyApple7]) {
@@ -239,15 +278,21 @@
 }
 
 void GrMtlCaps::initGPUFamily(id<MTLDevice> device) {
-    if (@available(macOS 10.15, iOS 13.0, tvOS 13.0, *)) {
+#if !defined(TARGET_OS_TV) 
+	if (@available(macOS 10.15, iOS 13.0, tvOS 13.0, *)) 
+#endif	
+	{
         if (this->getGPUFamily(device, &fGPUFamily, &fFamilyGroup)) {
             return;
         }
-    } else {
+    }
+#if !defined(TARGET_OS_TV) 	
+	else {
         if (this->getGPUFamilyFromFeatureSet(device, &fGPUFamily, &fFamilyGroup)) {
             return;
         }
     }
+#endif
     // We don't know what this is, fall back to minimum defaults
 #ifdef SK_BUILD_FOR_MAC
     fGPUFamily = GPUFamily::kMac;
diff --git a/src/gpu/ganesh/mtl/GrMtlCommandBuffer.mm b/src/gpu/ganesh/mtl/GrMtlCommandBuffer.mm
index 3bd7a192c1..df51a54d4b 100644
--- a/src/gpu/ganesh/mtl/GrMtlCommandBuffer.mm
+++ b/src/gpu/ganesh/mtl/GrMtlCommandBuffer.mm
@@ -23,7 +23,7 @@
 
 sk_sp<GrMtlCommandBuffer> GrMtlCommandBuffer::Make(id<MTLCommandQueue> queue) {
     id<MTLCommandBuffer> mtlCommandBuffer;
-#if GR_METAL_SDK_VERSION >= 230
+#if GR_METAL_SDK_VERSION >= 230 && !(defined(TARGET_OS_TV))
     if (@available(macOS 11.0, iOS 14.0, tvOS 14.0, *)) {
         MTLCommandBufferDescriptor* desc = [[MTLCommandBufferDescriptor alloc] init];
         desc.errorOptions = MTLCommandBufferErrorOptionEncoderExecutionStatus;
diff --git a/src/gpu/ganesh/mtl/GrMtlGpu.mm b/src/gpu/ganesh/mtl/GrMtlGpu.mm
index 0cac6cb18e..1862e55e0a 100644
--- a/src/gpu/ganesh/mtl/GrMtlGpu.mm
+++ b/src/gpu/ganesh/mtl/GrMtlGpu.mm
@@ -1815,10 +1815,12 @@ void copy_src_data(char* dst,
     }
 #endif  // SK_BUILD_FOR_MAC
 #if __MAC_OS_X_VERSION_MAX_ALLOWED >= 101500 || __IPHONE_OS_VERSION_MAX_ALLOWED >= 130000
+#if !(defined(TARGET_OS_TV) && TARGET_OS_TV)
     if (@available(macOS 10.15, iOS 13.0, tvOS 13.0, *)) {
         writer->appendBool("hasUnifiedMemory", fDevice.hasUnifiedMemory);
     }
 #endif
+#endif
 #ifdef SK_BUILD_FOR_MAC
 #if __MAC_OS_X_VERSION_MAX_ALLOWED >= 101500
     if (@available(macOS 10.15, *)) {
@@ -1899,9 +1901,11 @@ void copy_src_data(char* dst,
         writer->appendU64("maxArgumentBufferSamplerCount", fDevice.maxArgumentBufferSamplerCount);
     }
 #ifdef SK_BUILD_FOR_IOS
+#if !(defined(TARGET_OS_TV) && TARGET_OS_TV)
     if (@available(iOS 13.0, tvOS 13.0, *)) {
         writer->appendU64("sparseTileSizeInBytes", fDevice.sparseTileSizeInBytes);
     }
+#endif
 #endif
     writer->endObject();
 
diff --git a/src/gpu/graphite/mtl/MtlCaps.mm b/src/gpu/graphite/mtl/MtlCaps.mm
index 871575ff10..63b5de5514 100644
--- a/src/gpu/graphite/mtl/MtlCaps.mm
+++ b/src/gpu/graphite/mtl/MtlCaps.mm
@@ -40,6 +40,7 @@
 }
 
 bool MtlCaps::GetGPUFamily(id<MTLDevice> device, GPUFamily* gpuFamily, int* group) {
+#if !(defined(TARGET_OS_TV) && TARGET_OS_TV)
 #if SKGPU_GRAPHITE_METAL_SDK_VERSION >= 220
     if (@available(macOS 10.15, iOS 13.0, tvOS 13.0, *)) {
         // Apple Silicon
@@ -50,6 +51,7 @@
             return true;
         }
 #endif
+#if !(defined(TARGET_OS_TV) && TARGET_OS_TV)
 #ifdef SK_BUILD_FOR_IOS
         if ([device supportsFamily:MTLGPUFamilyApple6]) {
             *gpuFamily = GPUFamily::kApple;
@@ -82,6 +84,7 @@
             return true;
         }
 #endif
+#endif //tvos
 
         // Older Macs
         // MTLGPUFamilyMac1, MTLGPUFamilyMacCatalyst1, and MTLGPUFamilyMacCatalyst2 are deprecated.
@@ -101,6 +104,7 @@
         }
     }
 #endif
+#endif //tvos
 
     // No supported GPU families were found
     return false;
